### DISCUSSION TITLE
First Invocation Of Parse_Latex Takes 10+ Seconds

### DISCUSSION THREAD

**Discussion by gerryfletch** [sympy/sympy#24911] [Q&A]
Hello,

I run some simple Sympy calculations as part of a Lambda. The first step is to parse some user supplied LaTeX, such as `\\left(x+1\\right)\\left(x+3\\right)`. On the first invocation, this usually takes upwards of 10 seconds. Subsequent calls with both the same input, and different input, is very quick at anywhere between 50-200 milliseconds.

Is this because parts of the Sympy framework are lazy loaded? Because a cache hasn't been loaded? What can I do to improve the first-time invocation time?

**Comment by ktunprasert** [comment]
I'm also experiencing the same issues running Python 3.8 Lambda on AWS EU-West

**Comment by asmeurer** [comment]
I don't see this behavior, at least in sympy master. It does have to load antlr and the grammar, but I wouldn't expect that to take that long. For me the first run is slower, but it's more on the order of about 600 ms compared to 12 ms for subsequent runs. I think we might have updated the antlr runtime version since the last release so you can also try git master or the 1.12 release candidate with the newer version of the antlr runtime to see if it still has the issue. 

Can you try running it through a profiler like cProfile or pyinstrument to see where it is spending the time? You may also get an idea just from keyboard interrupting in the middle of the first call and looking at the resulting traceback.

**Reply by gerryfletch** [reply]
Thanks, helpful to know that it shouldn't be that slow - I'll double check versions and look into profiling it :)
